### PR TITLE
[PLUGIN-370] Improved Cassandra Validations

### PR DESCRIPTION
**Pull request by dli357**
See [PLUGIN-370](https://issues.cask.co/browse/PLUGIN-370). Moved some validations to validate function, ensure validations on username+password is done in both configurePipeline and prepareRun.

### CODE REVIEW

File: cassandra-plugins/src/main/java/io/cdap/plugin/batch/source/BatchCassandraSource.java [hunk 1]
```diff
@@ -273,6 +272,22 @@ private Schema getSchema() {
...
+
+    public void validate(FailureCollector collector) {
+      if (Strings.isNullOrEmpty(schema)) {
+        collector.addFailure("Schema must be specified.", null).withConfigProperty(Cassandra.SCHEMA);
+      }
+      if (port != null && (port < 1 || port > 65535)) {
+        collector.addFailure("Invalid port.", "Port number must be an integer from 1 to 65535.")
+          .withConfigProperty(BatchCassandraSink.Cassandra.PORT);
+      }
+      if ((!containsMacro(Cassandra.USERNAME) && !containsMacro(Cassandra.PASSWORD)) &&
```
Comment (shifuxu0301) [inline cassandra-plugins/src/main/java/io/cdap/plugin/batch/source/BatchCassandraSource.java]: more clear to put in this way:

!containsMacro(Cassandra.USERNAME) && !containsMacro(Cassandra.PASSWORD) && (Strings.isNullOrEmpty(username) ^ Strings.isNullOrEmpty(password))

File: cassandra-plugins/src/main/java/io/cdap/plugin/batch/sink/BatchCassandraSink.java [hunk 2]
```diff
@@ -178,6 +189,13 @@ public CassandraBatchConfig(String referenceName, String partitioner, @Nullable
       this.columns = columns;
       this.primaryKey = primaryKey;
     }
+
+    public void validate(FailureCollector collector) {
+      if (port != null && (port < 1 || port > 65535)) {
+        collector.addFailure("Invalid port.", "Port number must be an integer from 1 to 65535.")
```
Comment (shifuxu0301) [inline cassandra-plugins/src/main/java/io/cdap/plugin/batch/sink/BatchCassandraSink.java]: "Invalid port number." Same thing below.

**Review by shifuxu0301 — APPROVED** [review]
Overall LGTM, some nit comments.

File: cassandra-plugins/src/main/java/io/cdap/plugin/batch/source/BatchCassandraSource.java (hunk 1 above)
Comment (dli357) [reply cassandra-plugins/src/main/java/io/cdap/plugin/batch/source/BatchCassandraSource.java]: Fixed

File: cassandra-plugins/src/main/java/io/cdap/plugin/batch/sink/BatchCassandraSink.java (hunk 2 above)
Comment (dli357) [reply cassandra-plugins/src/main/java/io/cdap/plugin/batch/sink/BatchCassandraSink.java]: Fixed

**Pull request merged**